node/node.go

```diff
@@ -76,60 +71,40 @@ func New(messageService messageservice.MessageService, chainservice chainservice
 
 	n.channelNotifier = notifier.NewChannelNotifier(store, n.vm)
 
-	// Start the engine in a go routine
-	ctx, cancel := context.WithCancel(context.Background())
-
-	n.wg = &sync.WaitGroup{}
-	n.wg.Add(1)
-
-	n.cancelEventHandler = cancel
-	// Start the event handler in a go routine
-	// It will listen for events from the engine and dispatch events to node channels
-	go n.handleEngineEvents(ctx)
-
 	return n
 }
 
-// handleEngineEvents is responsible for monitoring the ToApi channel on the engine.
-// It parses events from the ToApi chan and then dispatches events to the necessary node chan.
-func (n *Node) handleEngineEvents(ctx context.Context) {
-	for {
+// handleEngineEvents dispatches events to the necessary node chan.
+func (n *Node) handleEngineEvent(update engine.EngineEvent) {
+	for _, completed := range update.CompletedObjectives {
+		d, _ := n.completedObjectives.LoadOrStore(string(completed.Id()), make(chan struct{}))
+		close(d)
+
+		// use a nonblocking send to the RPC Client in case no one is listening
```

Comment:
Outside the scope of this PR, but a helper function to handle the nonblocking send might make things slightly neater.